Meshtastic/Helpers/BLEManager.swift

```diff
@@ -897,41 +905,50 @@ class BLEManager: NSObject, CBPeripheralDelegate, MqttClientProxyManagerDelegate
-								hopNode = createNodeInfo(num: Int64(node), context: context)
-							}
-							let traceRouteHop = TraceRouteHopEntity(context: context)
-							traceRouteHop.time = Date()
-							traceRouteHop.back = true
-							if routingMessage.snrBack.count >= index + 1 {
-								traceRouteHop.snr = Float(routingMessage.snrBack[index] / 4)
-							}
-							if let hn = hopNode, hn.hasPositions {
-								if let mostRecent = hn.positions?.lastObject as? PositionEntity, mostRecent.time! >= Calendar.current.date(byAdding: .hour, value: -24, to: Date())! {
-									traceRouteHop.altitude = mostRecent.altitude
-									traceRouteHop.latitudeI = mostRecent.latitudeI
-									traceRouteHop.longitudeI = mostRecent.longitudeI
-									traceRoute?.hasPositions = true
+						// Only if hopStart is set and there is an SNR entry
+						if decodedInfo.packet.hopStart > 0 && routingMessage.snrBack.count > 0 {
+							var routeBackString = "\(traceRoute?.node?.user?.longName ?? "unknown".localized) \((traceRoute?.node?.num ?? 0).toHex()) --> "
+							for (index, node) in routingMessage.routeBack.enumerated() {
+								var hopNode = getNodeInfo(id: Int64(node), context: context)
+								if hopNode == nil && hopNode?.num ?? 0 > 0 && node != 4294967295 {
+									hopNode = createNodeInfo(num: Int64(node), context: context)
 								}
-							}
-							traceRouteHop.num = hopNode?.num ?? 0
-							if hopNode != nil {
-								if decodedInfo.packet.rxTime > 0 {
-									hopNode?.lastHeard = Date(timeIntervalSince1970: TimeInterval(Int64(decodedInfo.packet.rxTime)))
+								let traceRouteHop = TraceRouteHopEntity(context: context)
+								traceRouteHop.time = Date()
+								traceRouteHop.back = true
+								if routingMessage.snrBack.count >= index + 1 {
+									traceRouteHop.snr = Float(routingMessage.snrBack[index]) / 4
+								} else {
+									// If no snr in route, set to unknown
+									traceRouteHop.snr = -32
 								}
+								if let hn = hopNode, hn.hasPositions {
+									if let mostRecent = hn.positions?.lastObject as? PositionEntity, mostRecent.time! >= Calendar.current.date(byAdding: .hour, value: -24, to: Date())! {
+										traceRouteHop.altitude = mostRecent.altitude
+										traceRouteHop.latitudeI = mostRecent.latitudeI
+										traceRouteHop.longitudeI = mostRecent.longitudeI
+										traceRoute?.hasPositions = true
+									}
+								}
+								traceRouteHop.num = hopNode?.num ?? 0
+								if hopNode != nil {
+									if decodedInfo.packet.rxTime > 0 {
+										hopNode?.lastHeard = Date(timeIntervalSince1970: TimeInterval(Int64(decodedInfo.packet.rxTime)))
+									}
+								}
+								hopNodes.append(traceRouteHop)
+								routeBackString += "\(hopNode?.user?.longName ?? String(hopNode?.num.toHex() ?? "unknown".localized)) \(hopNode?.viaMqtt ?? false ? "MQTT" : "") (\(traceRouteHop.snr != -32 ? String(traceRouteHop.snr) : "unknown ".localized)dB) --> "
```

Comment:
same nit about formatting + whitespace for mqtt string

Reply:
Also done.